pkg/controllers/leaderworkerset_controller.go

```diff
@@ -306,10 +310,13 @@ func (r *LeaderWorkerSetReconciler) rollingUpdateParameters(ctx context.Context,
 
 	// When updated replicas become not ready again or scaled up replicas are not ready yet,
 	// we'll not modify the Partition field. That means Partition moves in one direction to make it simple.
+	if partition > utils.NonZeroValue(stsReplicas-int32(rollingStep)-continuousReadyReplicas) {
+		r.Record.Eventf(lws, corev1.EventTypeNormal, "GroupsAreUpgrading", fmt.Sprintf("changing partition from %d, to %d", partition, utils.NonZeroValue(stsReplicas-int32(rollingStep)-continuousReadyReplicas)))
+	}
 	return min(partition, utils.NonZeroValue(stsReplicas-int32(rollingStep)-continuousReadyReplicas)), wantReplicas(lwsUnreadyReplicas), nil
 }
 
-func (r *LeaderWorkerSetReconciler) SSAWithStatefulset(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, partition, replicas int32, revisionKey string) error {
+func (r *LeaderWorkerSetReconciler) SSAWithStatefulset(ctx context.Context, lws *leaderworkerset.LeaderWorkerSet, leaderSts *appsv1.StatefulSet, partition, replicas int32, revisionKey string) error {
```

Comment:
instead of passing the leaderSts just to check if it is nil, lets record the event outside the function. Also, lets emit an event if we fail to create the sts.